pkg/dss/geo/s2.go

```diff
@@ -84,10 +91,40 @@ func Volume3DToCellIDs(v3 *models.Volume3D) (s2.CellUnion, error) {
 	if v3 == nil {
 		return nil, errBadCoordSet
 	}
-	if v3.PolygonFootprint == nil {
+
+	switch t := v3.Footprint.(type) {
```

Comment:
Rather than switching here, could we have ToCellIDs be a member function of the Geometry interface?

Reply:
I started out with that approach, but found that it breaks encapsulation quite a lot and I would rather revisit the idea in a follow-up PR if that's okay with you.